src/lib/db.js

```diff
@@ -4,35 +4,64 @@ import _ from 'lodash';
 
 let _db = undefined;
 
-export class Activity {
+// In the future, use private fields instead:
+//   https://github.com/tc39/proposal-class-fields
+// I thought I could use symbols, but that didn't work out.
+let modelAttrs = {};
+
+class Model {
+  constructor(table, key) {
+    modelAttrs[this.constructor] = { table: table, key: key };
+  }
+
+  async save() {
+    // Does an update if the row already exists, otherwise does a put.
+    let keyname = modelAttrs[this.constructor].key;
+    let key = this[keyname];
+    let table = modelAttrs[this.constructor].table;
+    const exists =
+      (await table
+        .where(keyname)
+        .equals(this[keyname])
+        .count()) > 0;
+    if (exists) {
+      return table.update(key, this);
+    } else {
+      return this.put();
```

Comment:
If you're going with this solution couldn't `put` be an `add`?

Reply:
Yeah, but did as you suggested.